File: findSumPair.js
function findSumPair(numbers, k) {
    /**
     * Find the indices of two integers in the array that sum to exactly k.
     * 
     * @param {number[]} numbers - Array of positive integers
     * @param {number} k - Target sum (positive integer)
     * @returns {number[]} Array of two indices [i, j] where numbers[i] + numbers[j] = k, or empty array if no pair found
     */
    
    // Use a Map to store numbers we've seen and their indices
    const seen = new Map();
    
    for (let i = 0; i < numbers.length; i++) {
        const num = numbers[i];
        const complement = k - num;
        
        // If we've seen the complement before, we found a pair
        if (seen.has(complement)) {
            return [seen.get(complement), i];
        }
        
        // Add current number and its index to seen map
        seen.set(num, i);
    }
    
    return [];
}

// Test with the provided example
const numbers = [1, 5, 8, 1, 2];
const k = 13;

const result = findSumPair(numbers, k);
console.log(`numbers = [${numbers}]`);
console.log(`k = ${k}`);
console.log(`Result: ${result}`);

// Additional test cases
console.log('\nAdditional tests:');
console.log(`findSumPair([1, 2, 3, 4], 7): ${findSumPair([1, 2, 3, 4], 7)}`);  // true (3+4)
console.log(`findSumPair([1, 2, 3, 4], 10): ${findSumPair([1, 2, 3, 4], 10)}`);  // false
console.log(`findSumPair([5, 5], 10): ${findSumPair([5, 5], 10)}`);  // true (5+5)

// Export for use in other modules
if (typeof module !== 'undefined' && module.exports) {
    module.exports = findSumPair;
}

File: archived_tests/find_sum_pair.py
def findSumPair(numbers, k):
    """
    Find if there exists a pair of integers in the array that sum to exactly k.
    
    Args:
        numbers: List of positive integers
        k: Target sum (positive integer)
    
    Returns:
        bool: True if a pair exists that sums to k, False otherwise
    """
    # Use a set to store numbers we've seen for O(1) lookup
    seen = set()
    
    for num in numbers:
        # Calculate what number we need to complete the sum
        complement = k - num
        
        # If we've seen the complement before, we found a pair
        if complement in seen:
            return True
        
        # Add current number to seen set
        seen.add(num)
    
    return False


# Test with the provided example
if __name__ == "__main__":
    numbers = [1, 5, 8, 1, 2]
    k = 13
    
    result = findSumPair(numbers, k)
    print(f"numbers = {numbers}")
    print(f"k = {k}")
    print(f"Result: {result}")
    
    # Additional test cases
    print("\nAdditional tests:")
    print(f"findSumPair([1, 2, 3, 4], 7): {findSumPair([1, 2, 3, 4], 7)}")  # True (3+4)
    print(f"findSumPair([1, 2, 3, 4], 10): {findSumPair([1, 2, 3, 4], 10)}")  # False
    print(f"findSumPair([5, 5], 10): {findSumPair([5, 5], 10)}")  # True (5+5)